src/cow-react/modules/limitOrders/hooks/useGetInitialPrice.ts

```diff
@@ -21,15 +18,22 @@ async function requestPriceForCurrency(chainId: number | undefined, currency: Cu
   }
 
   if (currency.isNative) {
-    return new BigNumber(1)
+    return 1
   }
 
   try {
     const result = await getNativePrice(chainId, currencyAddress)
 
-    const price = result ? adjustDecimals(result?.price, currency.decimals) : undefined
+    if (!result) {
+      throw new Error('Cannot parse initial price')
+    }
 
-    if (!price) return new Error('Cannot parse initial price')
+    // TODO: IS THIS CORRECT? (I've used 18 here as arbitrary number to remove decimals)
+    const price = result.price * 10 ** (18 + getDecimals(currency))
```

Comment:
Yes, it looks correct, because it's a price in ETH (native token) units. Just move 18 to const: `const ETH_DECIMALS = 18`